keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,960 @@
+  - [Custom CSI Plugin](#custom-csi-plugin)
+  - [Advantages of In-Tree OCI VolumeSource](#advantages-of-in-tree-oci-volumesource)
+  - [Conclusion](#conclusion)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The proposed enhancement adds a new `VolumeSource` to Kubernetes that supports OCI images and/or OCI artifacts.
+This allows users to package files and share them among containers in a pod without including them in the main image,
+thereby reducing vulnerabilities and simplifying image creation.
```

Comment:
Adding extra package sourcing would result in an increase in attack surface. Stating as a conclusion that `VolumeSource`s will reduce vulnerabilities might therefore not be the case. Unless`VolumeSource`s isn't used in conjunction with image repos. A combination of both might result in higher vulnerability.

Reply:
Thanks for bringing this up - I agree the Summary is not as clearly stated as it should be - I'll update it - my thinking is more this: 
```
Introducing a new VolumeSource does not inherently reduce vulnerabilities, but design choices around read-only,
non-executable mounts, and the decoupling of application and data can contribute to a more secure overall system.
The goal is to provide a secure, modular, and flexible way to manage files (most importantly today but not limited to 
model files), that aligns with best practices for container security. For example, a model-server can switch between
different models without altering model-server container. This flexibility simplifies updates and allows for quick
changes without requiring a full container rebuild.
```